hotsos/client.py

```diff
@@ -265,14 +262,25 @@ def get(self, format=None, html_escape=False, minimal_mode=None,
         if minimal_mode:
             filtered = self.minimise(filtered, minimal_mode)
 
-        if format == 'json':
-            log.debug('Converting master yaml file to %s', format)
-            filtered = json.dumps(filtered, indent=2, sort_keys=True)
-        else:
+        if format is None:
+            format = 'yaml'
+
```

Comment:
mutable values for defaults are considered bad practice so i try to avoid them. this is mainly because they can easily be permanently changed so that subsequent calls get different defaults e.g.

```
def foo(arg=[1]):
    arg.pop()
````
having said that since a string is immutable we could safely use it as a default so I will put it back.

Reply:
Ok, that is subtle. I hadn't realized that Python evaluates default arguments only once. Seems like a bit of a design flaw 😄